mne/utils.py

```diff
@@ -510,6 +510,22 @@ def _reject_data_segments(data, reject, flat, decim, info, tstep):
     return data, drop_inds
 
 
+def _get_inst_data(inst):
+    """get data from MNE object instance like Raw, Epochs or Evoked.
+    Returns a view, not a copy!"""
+    from .io.base import _BaseRaw
+    from .epochs import _BaseEpochs
+    from . import Evoked
+    from .time_frequency.tfr import _BaseTFR
+
+    if isinstance(inst, (_BaseRaw, _BaseEpochs)):
+        if not inst.preload:
+            inst.load_data()
```

Comment:
I can do that but I am not sure what is better - you get the data anyhow now (so they take up memory) and the data are not copied (so you don't have two copies of the same data).